src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,46 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter DATE_TIME_FORMATTER =
+            DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        if (names == null || data == null || dateFrom.isEmpty() || dateTo.isEmpty()) {
+            throw new IllegalArgumentException("Dates or data cannot be null or empty");
+        }
+
+        LocalDate startDate = LocalDate.parse(dateFrom, DATE_TIME_FORMATTER);
+        LocalDate endDate = LocalDate.parse(dateTo, DATE_TIME_FORMATTER);
+
+        StringBuilder sb = new StringBuilder("Report for period "
+                + dateFrom + " - " + dateTo + System.lineSeparator());
+
+        for (String name : names) {
+            int salary = 0;
+
+            for (String info : data) {
+                String[] parts = info.split(" ");
+                String dateFromData = parts[0];
```

Comment:
add constants for indexes
```suggestion
                String dateFromData = parts[DATE_INDEX];
```